contracts/swap/OneInch.sol

```diff
@@ -0,0 +1,142 @@
+// SPDX-License-Identifier: BUSL-1.1
+pragma solidity 0.7.6;
+pragma experimental ABIEncoderV2;
+
+import "./BaseSwap.sol";
+import "@openzeppelin/contracts/token/ERC20/SafeERC20.sol";
+import "@openzeppelin/contracts/utils/Address.sol";
+import "@kyber.network/utils-sc/contracts/IERC20Ext.sol";
+import "../interfaces/IAggregationRouter.sol";
+
+contract OneInch is BaseSwap {
+    using SafeERC20 for IERC20Ext;
+    using Address for address;
+
+    IAggregationRouter public router;
+
+    event UpdatedAggregationRouter(IAggregationRouter router);
+
+    constructor(address _admin, IAggregationRouter _router) BaseSwap(_admin) {
+        router = _router;
+    }
+
+    function updateAggregationRouter(IAggregationRouter _router) external onlyAdmin {
+        router = _router;
+        emit UpdatedAggregationRouter(router);
+    }
+
+    /// @dev get expected return and conversion rate if using a Uni router
+    function getExpectedReturn(GetExpectedReturnParams calldata params)
+        external
+        view
+        override
+        onlyProxyContract
+        returns (uint256 destAmount)
+    {
+        require(false, "getExpectedReturn_notSupported");
+    }
+
+    function getExpectedIn(GetExpectedInParams calldata params)
+        external
+        view
+        override
+        onlyProxyContract
+        returns (uint256 srcAmount)
+    {
+        require(false, "getExpectedIn_notSupported");
+    }
+
+    /// @dev swap token
+    /// @notice
+    /// 1inch API will returns data neccessary to build tx
+    /// tx's data will be passed by params.extraData
+    function swap(SwapParams calldata params)
+        external
+        payable
+        override
+        onlyProxyContract
+        returns (uint256 destAmount)
+    {
+        require(params.tradePath.length == 2, "oneInch_invalidTradepath");
+
+        safeApproveAllowance(address(router), IERC20Ext(params.tradePath[0]));
+
+        bytes4 methodId = params.extraArgs[0] |
+            (bytes4(params.extraArgs[1]) >> 8) |
+            (bytes4(params.extraArgs[2]) >> 16) |
+            (bytes4(params.extraArgs[3]) >> 24);
```

Comment:
Why is this the methodId?

Reply:
We just define our extraArgs that way. Basically we get whatever returned from 1inch API `tx.data` and pass it to `params.extraArgs`